### ISSUE TITLE
How to force full re-initialisation of the list ?

### ISSUE THREAD

**Issue by dclause** [tangbc/vue-virtual-scroll-list#179]
Hi !

I use this component to render complex items in a list. I want to have multiple layouts of components depending on user preference: "compact list view", "dense list view", "standard list", "expanded list", etc..

I use the component as such :
```
<virtual-list
        class="virtual-list"
        :keeps="30"
        :data-key="'bid'"
        :data-sources="bookmarks"
        :data-component="itemComponent"
/>
```
`itemComponent` is a method that return the appropriate component depending on the display. To make it easy : 
```
itemComponent() {
      if (this.displayType === 'compact') return BookmarkCompactItem
      if (this.displayType === 'list') return BookmarkListItem
      if (this.displayType === 'card') return BookmarkCardItem
....
```

I have a watcher that detects the change on layout. I wonder how to reset the component on that occasion, ie go back to top + re-render the items.

Could you help me on that ?

- If I don't scroll at all on the page, then it works fine. But if I start scrolling (I guess the range changes), then any time I change the layout, the list gets blank.
- I have set a `:ref="displayStyle"` on the list, but changing the ref does not re-render the item (the list still gets blank).
- I have tried .$forceUpdate() on the component with no more success (still get that blank list)
- I have tried to remove the `oldValue.length != newValue.length` in the source code of the data.sources watcher, with still no effect.

Any advice on this ?

**Comment by tangbc** [comment]
Is working well by assign `data-component` as a function now ?  

Inside use an instance `this.virtual`, all state keep here, so here can offer a `reset()` method called via `ref`.

**Comment by tangbc** [comment]
This shoud be relate with #161

**Comment by tangbc** [comment]
Published in v2.0.5

See: https://github.com/tangbc/vue-virtual-scroll-list/tree/v2.0.5#public-methods
